File: examples_android/app/src/main/java/com/confirmit/testsdkapp/layouts/simpleCardLayout/items/SimpleCardNumericItem.kt
package com.confirmit.testsdkapp.layouts.simplecardlayout.items

import android.view.View
import android.widget.EditText
import com.confirmit.mobilesdk.ui.questions.NumericQuestion
import com.confirmit.testsdkapp.R

class SimpleCardNumericItem(view: View) : SimpleCardItem(view) {

    private val txtNumeric: EditText = view.findViewById(R.id.editText)
    private lateinit var question: NumericQuestion

    fun setupQuestion(question: NumericQuestion) {
        this.question = question
        txtHeader.text = question.text.get()
        txtNumeric.setText(question.getValue())

        handleValidation(question.errors)
    }

    override fun onShow(): Boolean {
        txtNumeric.requestFocus()
        return false
    }

    override fun onNext(): Boolean {
        txtNumeric.text.toString().toDoubleOrNull()?.let {
            question.setValue(it)
        }

        return validate(question)
    }

    override fun onBack(): Boolean {
        txtNumeric.text.toString().toDoubleOrNull()?.let {
            question.setValue(it)
        }

        return true
    }
}

File: examples_ios/MyApp/Sources/SurveyLayouts/SimpleCardLayout/Items/SimpleCardNumericItem.swift
import Foundation
import ConfirmitMobileSDK
import UIKit

class SimpleCardNumericItem: SimpleCardItem {
    @IBOutlet var txtNumeric: UITextField!
    
    var question: NumericQuestion!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        txtNumeric.tintColor = SimpleCardColor.button
        txtNumeric.font = UIFont.systemFont(ofSize: 16)
    }
    
    func setupQuestion(question: NumericQuestion) {
        self.question = question
        lblHeader.text = question.text.get()
        txtNumeric.text = question.getValue()
        
        handleValidation(errors: question.errors)
    }
    
    override func onShow() -> Bool {
        txtNumeric.becomeFirstResponder()
        return false
    }
    
    override func onNext() -> Bool {
        if let double = Double(txtNumeric.text ?? "") {
            question.setValue(value: double)
        }
        return validate(question: question)
    }
    
    override func onBack() -> Bool {
        if let double = Double(txtNumeric.text ?? "") {
            question.setValue(value: double)
        }
        return true
    }
}
